From fd6e7c8949218df6b61a5b9186aba3f9d1d91f75 Mon Sep 17 00:00:00 2001
From: Mario Limonciello <superm1@gmail.com>
Date: Mon, 9 May 2022 20:16:26 -0500
Subject: [PATCH] Set autosuspend delay for Fibocom LG850-GL

It has been shown that the autosuspend delay for this device enacted
by modem manager will race with suspend and cause system suspend
failures.

This occurred in ChromiumOS on a chromebook, but there is no reason
it won't happen in regular notebooks with the same WWAN.  To avoid
the failure delay autosuspend to a frequency longer than the polling
rate used by modem manager.

Link: https://chromium-review.googlesource.com/c/chromiumos/overlays/board-overlays/+/3635003
Link: https://github.com/freedesktop/ModemManager/blob/43e76bf1bbca8207e37d3fbddfed4bb8578f0170/src/mm-iface-modem.c#L1633
(cherry picked from commit f32049eda3b06deaae3d6d7d06a9dfd16528c443)

Related: #2087778
---
 hwdb.d/60-autosuspend.hwdb | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/hwdb.d/60-autosuspend.hwdb b/hwdb.d/60-autosuspend.hwdb
index 323611d000..07f7fa9344 100644
--- a/hwdb.d/60-autosuspend.hwdb
+++ b/hwdb.d/60-autosuspend.hwdb
@@ -75,13 +75,18 @@ usb:v0627p0001:*QEMU USB Tablet*
  ID_AUTOSUSPEND=1
 
 #########################################
-# Sierra Wireless
+# WWAN
 #########################################
 
 # Sierra Wireless EM7345 4G LTE modem
 usb:v1199pA001*
  ID_AUTOSUSPEND=1
 
+# Fibocom LG850-GL
+usb:v2CB7p0007*
+ ID_AUTOSUSPEND=1
+ ID_AUTOSUSPEND_DELAY_MS=7000
+
 #########################################
 # Wacom
 #########################################
